File: app/src/main/java/moe/notify/animenotifier/domain/model/anime/AnimeOpening.java
package moe.notify.animenotifier.domain.model.anime;


public class AnimeOpening {

    public float similarity;
    public String permalink;
    public String uri;
    public String title;
    public int likes;
    public int plays;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof AnimeOpening)) return false;

        AnimeOpening that = (AnimeOpening) o;

        if (!uri.equals(that.uri)) return false;
        return title.equals(that.title);

    }

    @Override
    public int hashCode() {
        int result = uri.hashCode();
        result = (31 * result) + title.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "AnimeOpening{" +
                "title='" + title + '\'' +
                ", uri='" + uri + '\'' +
                ", permalink='" + permalink + '\'' +
                ", likes=" + likes +
                ", plays=" + plays +
                ", similarity=" + similarity +
                '}';
    }
}

File: app/src/main/java/moe/notify/animenotifier/domain/model/anime/AnimeOpening.kt
package moe.notify.animenotifier.domain.model.anime


import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class AnimeOpening(
    open var similarity: Double = 0.toDouble(),
    open var permalink: String? = null,
    @PrimaryKey
    open var uri: String? = null,
    open var title: String? = null,
    open var likes: Int = 0,
    open var plays: Int = 0) : RealmObject()